nilearn/maskers/tests/test_nifti_masker.py

```diff
@@ -24,6 +24,114 @@
 from nilearn.maskers.nifti_masker import _filter_and_mask
 
 
+def test_default_filter_parameter():
+    """Test that the default filter is None."""
+    masker = NiftiMasker()
+    assert masker.filter is None
+    assert "filter" not in masker.clean_kwargs
+
+
+def test_filter_parameter_propagation():
+    """Test that the filter parameter is correctly passed to signal.clean."""
+    rng = np.random.default_rng()
+    data = rng.random((9, 9, 9, 10))
+    img = Nifti1Image(data, np.eye(4))
+    mask = np.zeros((9, 9, 9), dtype="uint8")
+    mask[3:6, 3:6, 3:6] = 1  # assert the mask has non zero parameter
+    mask_img = Nifti1Image(mask, np.eye(4))
+    mask_img = Nifti1Image(mask, np.eye(4))
+
+    masker = NiftiMasker(
+        mask_img=mask_img, filter="cosine", t_r=2, high_pass=0.01
+    )
+    transformed_data = masker.fit_transform(img)
+
+    assert "filter" in masker.clean_kwargs
+    assert masker.clean_kwargs["filter"] == "cosine"
+    assert transformed_data.shape[1] == mask.sum()
```

Comment:
```suggestion
def test_filter_parameter_propagation(shape_3d_default, img_3d_rand_eye, affine_eye):
    """Test that the filter parameter is correctly passed to signal.clean."""
    mask = np.zeros(shape_3d_default)
    mask[3:6, 3:6, 3:6] = 1  # assert the mask has non zero parameter
    mask_img = Nifti1Image(mask, affine_eye)

    masker = NiftiMasker(
        mask_img=mask_img, filter="cosine", t_r=2, high_pass=0.01
    )
    transformed_data = masker.fit_transform(img_3d_rand_eye)

    assert "filter" in masker.clean_kwargs
    assert masker.clean_kwargs["filter"] == "cosine"
    assert transformed_data.shape[1] == mask.sum()
```


You can use test fixture (think of them data generated before each test) to make it easier to write tests.

Most of them are listed in `nilearn/conftest.py` (look for functions with the decorator `@pytest.fixture`)

See our contributing doc for more info: https://nilearn.github.io/dev/development.html#fixtures

Showing you how to modify this one but I will let you change the other ones